### ISSUE TITLE
[Feature Request] Option to how routing is done from a service to containers

### ISSUE THREAD

**Issue by Power2All** [docker/for-linux#1276]
Me and my colleague had been pondering on this for a while, and eventually we didn't find a solution for the current docker environment we work with. We do have some solutions, but those are kind of work-around and such, and thus we came with another conclusion that could help us, but perhaps also others.

We notice that Docker by default, when using services, uses a load-balancing way of routing traffic, this is perfect as it is, and works as expected, however, when using a fail-over networking with services, it doesn't work as expected. We would like to have a priority-way of working with routing in the networking, but we were not able to find such a solution.

Instead of load-balancing or round-robin connecting to containers in services, it would be nice to have a priority way of connecting to the containers. What I mean, is that a connection will ALWAYS goes to the first container, and only tries the other, in case a container is timing out of not accepting connections. Would be nice if such a feature would exist, so we could have a fail-over system for like Redis. We don't really care if it's not replicated, but we do want that all connections goes to one container, and tries another container (if deployed and running) when the first one fails.

Could anybody enlighten me if such a feature is already there, or if we overlooked perhaps something with this ?
Much appreciated !

* [ ] This is a bug report
* [x] This is a feature request
* [x] I searched existing issues before opening this one

### TLDR;

Creating a docker-compose file, in which we could set the way routing is done when dealing with multiple containers in a single service. Instead of "Round Robin", making it connect based on priority, or deployment time. I've seen that Docker uses a internal DNS handling system, and DNS is known for allowing priority list or "serial"-based connection (try first, and continue to the next, etc.., instead of parallel)
